core/src/main/java/net/sourceforge/jnlp/JNLPFile.java

```diff
@@ -223,72 +217,41 @@
             defaultArch = JavaSystemProperties.getOsArch();
         }
         catch (SecurityException ex) {
-            // null values will still work, and app can set defaults later
+            // FIXME: how should we proceed if the default values are not available??
         }
     }
 
     /**
      * Empty stub, allowing child classes to override the constructor
      */
+    // only used for tests
     protected JNLPFile() {
```

Comment:
why do we need this constructor?

Reply:
there are dummy sub-classes in tests which call this constructors